packages/common/src/services/auth/hedgehog.ts

```diff
@@ -175,5 +185,42 @@ export const createHedgehog = ({
     }
   }
 
+  const originalConfirmCredentials = hedgehog.confirmCredentials.bind(hedgehog)
+  hedgehog.confirmCredentials = async ({
+    softCheck,
+    ...confirmCredentialsArgs
+  }: ConfirmCredentialsArgs) => {
+    if (softCheck) {
+      const lookupKey = await getLookupKey({
+        username: confirmCredentialsArgs.username,
+        password: confirmCredentialsArgs.password
+      })
+      try {
+        const wallet = hedgehog.wallet
+        if (!wallet) {
+          throw new Error(
+            'Cannot check credentials - user is not authenticated'
+          )
+        }
+        const headers = await getAuthHeaders(wallet)
+        if (headers[AuthHeaders.Message] && headers[AuthHeaders.Signature]) {
+          return await makeIdentityRequest({
+            url: '/authentication/check',
+            method: 'GET',
+            headers,
+            params: { lookupKey }
+          })
+        } else {
+          throw new Error(
+            'Cannot check credentials - user is not authenticated'
+          )
+        }
+      } catch (e) {
```

Comment:
no need to log error here? this indicates user is not signed in?

Reply:
The thought is that you should not call `checkCredentials` if not signed in. But this code was mostly lifted from the libs implementation and cleaned up a little. I can take a closer look at how it's used. But I think it's mostly in the change password flows, which you should _definitely_ not be doing when logged out.